pkg/keylock/keylock.go

```diff
@@ -83,6 +92,31 @@ func (kl *keyLock) Acquire(id string) LockHandle {
 	return *h
 }
 
+func (kl *keyLock) AcquireWithTimeout(id string, duration time.Duration) (LockHandle, error) {
+	h := kl.getOrCreateLock(id)
+	timeout := time.After(duration)
+	for {
+		select {
+		case <-timeout:
+			h.refcnt--
+			return *h, fmt.Errorf("error waiting for lock")
+		default:
+			if kl.tryLock(id) {
+				h.genNum++
+				return *h, nil
+			}
+			time.Sleep(1 * time.Millisecond)
+		}
+	}
+
+}
+
+func (kl *keyLock) tryLock(id string) bool {
+	h := kl.getOrCreateLock(id)
+	return atomic.CompareAndSwapInt32((*int32)(unsafe.Pointer(h.mutex)), 0, mutexLocked)
```

Comment:
Just curious. Is this how trylock is implemented in go 1.18?

Reply:
done - https://portworx.atlassian.net/browse/PWX-31707